File: Study Plans/Binary Search/Java/Two-Sum-II-Input-Array-is-Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int low = 0;
        int high = numbers.length-1;
        
        while (low <= high) {
            
            int sum = numbers[low]+numbers[high];
            
            if (sum == target) {
                int[] res = new int[2];
                res[0] = low+1;
                res[1] = high+1;
                return res;
            }
            
            if (sum < target) {
                low++;
            } else {
                high--;
            }
        }
        
        return null;
    }
}

File: Study Plans/Binary Search/Python/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        low = 0
        high = len(numbers)-1
        
        while low <= high:
            
            res = numbers[low]+numbers[high]
            
            if res == target:
                return [low+1, high+1]
            
            if res < target:
                low +=1
            
            if res > target:
                high -=1
        
        
